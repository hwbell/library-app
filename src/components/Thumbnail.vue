<template>
  <!-- <img class="thumbnail" :src="this.url" alt="img"/> -->
  <transition name="fade">
    <div>
      <img
        class="thumbnail"
        :src="this.source.volumeInfo.imageLinks.thumbnail"
        alt="img"
        @click="showDetail"
      >
    </div>
  </transition>
</template>

<script>
export default {
  name: "Thumbnail",
  props: {
    source: {
      type: Object,
      required: true
    },
    type: {
      type: String,
      required: true
    }
  },
  methods: {
    showDetail() {
      console.log("emitting showBookDetail");
      this.$emit("showBookDetail", {
        book: this.source,
        type: this.type
      });
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
$link-blue: rgb(0, 119, 255);

.thumbnail {
  cursor: pointer;
  margin: 15px 10px;
  height: 150px;
  width: 100px;
  box-shadow: -2px 2px 5px 1px rgba(24, 24, 24, 0.2);
  transition: all 0.15s linear;
  &:hover {
    box-shadow: 0px 0px 5px 2px rgba(49, 42, 148, 0.52);
    -webkit-transform: translate(1px);
    -ms-transform: translate(1px);
    transform: translate(1px);
  }
}
// for the animation of the thumbnail
.fade-enter-active,
.fade-leave-active {
  transition: all 0.3s;
}
.fade-enter,
.fade-leave-to {
  opacity: 0;
}
</style>