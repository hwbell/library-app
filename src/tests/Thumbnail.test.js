// Import Vue and the component being tested
import { shallowMount, mount } from '@vue/test-utils'

import Vue from 'vue'
import Thumbnail from '../components/Thumbnail.vue';

describe('Thumbnail', () => {

  // define some props
  let someProps = {
    propsData: {
      source: {
        volumeInfo: {
          imageLinks: {
            thumbnail: 'https://books.google.com/books/content?id=antycYtRXOUC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api'
          }
        }
      },
      type: 'search'
    }
  }
  it('renders correctly', () => {
    const wrapper = mount(Thumbnail, someProps)
    expect(wrapper.element).toMatchSnapshot()
  })

  const wrapper = mount(Thumbnail, someProps)
  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('contains an image', () => {
    expect(wrapper.html()).toContain('img')
  })
  it('should fire the showBookDetail emitter', () => {

    // fire and check if it emits the right event
    wrapper.vm.showDetail() // should emit 'showBookDetail' to parent

    // assert event has been emitted
    expect(wrapper.emitted().showBookDetail).toBeTruthy()

  })


})

