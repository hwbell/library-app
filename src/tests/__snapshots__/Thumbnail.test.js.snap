// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders correctly 1`] = `
<div
  name="fade"
>
  <img
    alt="img"
    class="thumbnail"
    src="https://books.google.com/books/content?id=antycYtRXOUC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
  />
</div>
`;
